diff --git a/src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php b/src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php
index 0f33a7f..76bcea5 100644
--- a/src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php
+++ b/src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php
@@ -15,6 +15,7 @@ class Normalizer
         'any' => 'mixed',
         'anytype' => 'mixed',
         'anyxml' => 'string',
+        'anyuri' => 'string',
         'anysimpletype' => 'mixed',
         'long' => 'int',
         'short' => 'int',
