--- a/src/Encoder/ObjectEncoder.php
+++ b/src/Encoder/ObjectEncoder.php
@@ -103,7 +103,7 @@
                             $iso = $objectAccess->isos[$normalizePropertyName];

                             return match(true) {
-                                $isAttribute => $value ? (new AttributeBuilder(
+                                $isAttribute => $value !== null ? (new AttributeBuilder(
                                     $type,
                                     $iso->to($value)
                                 ))(...) : $defaultAction,
